src/NHibernate/Hql/Ast/ANTLR/Tree/FromElementType.cs

```diff
@@ -426,21 +426,10 @@ public string[] ToColumns(string tableAlias, string path, bool inSelect, bool fo
 					//              B) otherwise, we need to use the persister's
 					//                  table name as the column qualification
 					//      2) otherwise (not correlated), use the given alias
-					if (IsCorrelation)
-					{
-						if (IsMultiTable)
-						{
-							return propertyMapping.ToColumns(tableAlias, path);
-						}
-						else
-						{
-							return propertyMapping.ToColumns(ExtractTableName(), path);
-						}
-					}
-					else
-					{
-						return propertyMapping.ToColumns(tableAlias, path);
-					}
+					if (_fromElement.Walker.StatementType == HqlSqlWalker.UPDATE && IsCorrelation)
```

Comment:
With the latest changes the comment above is outdated.

Reply:
Latest changes are incorrect.. I will adjust comment when it's ready